django_feature_flip/helpers.py

```diff
@@ -1,22 +1,90 @@
 
-from django_feature_flip.models import Feature
+"""Main helper file of FeatureFlip."""
+
+from django_feature_flip.models import Group, Feature
 
 
 class FeatureFlip:
+    """Main class, singleton that should always be instanciated as FeatureFlip()."""
+
+    class FeatureFlipError(Exception):
+        """Default class for the FeatureFlip errors."""
+
+        def __init__(self, message):
+            """Initialize instance."""
+            self.message = message
+
     class __FeatureFlip:
+        """Initializer class for the singleton of FeatureFlip."""
+
         def __init__(self):
-            pass
+            """Initialize instance."""
+            self.groups = {}
 
         def __str__(self):
             return repr(self)
     instance = None
 
     def __init__(self):
+        """Initialize instance."""
         if not FeatureFlip.instance:
             FeatureFlip.instance = FeatureFlip.__FeatureFlip()
 
     def __getattr__(self, name):
+        """Get attribute."""
         return getattr(self.instance, name)
 
     def enabled(self, feature_name, actor=None):
```

Comment:
```suggestion
    def enabled(self, feature_name, actor=None) -> bool:
```
This kind of function seems like a good place to start adding type annotations.
Not only they document the code, but we can check it later with [flare](https://pyre-check.org/index.html)

Reply:
Quick question as to typing the return value... do we have to type the returned variable too?

Reply:
i think it shouldn't be neccessary since the return type can be inferred from the arguments and the calls return types